### DISCUSSION TITLE
Arduino OTA using USBASP or JTAG

### DISCUSSION THREAD

**Discussion by basitalikhan** [JAndrassy/ArduinoOTA#87] [General]
I have successfully done OTA in the following way:

USBASP -> upload bootloader -> then Upload Code using Serial lines through USB.

But if I upload code like:

JTAG/USBASP -> Code File -with bootloader using JTAG or USBASP

The code then hangs and the same code does not respond. What should I do?

**Comment by JAndrassy** [comment]
where is the ArduinoOTA or any OTA in described uploading?

**Reply by basitalikhan** [reply]
Sorry I confused you.. I will come on to the problem... If I upload my code using USBASP or JTAG with the code file having bootloader and then I download the file, it does not perform the OTA, the MCU hangs. The MCU is ATMEGA2560.

**Reply by JAndrassy** [reply]
and do you burn the sketch with the right _with_bootloader? with a board in Tools selected which has Optiboot with copy_flash_pages function? my_boards or MCUDude core?
